src/entrypoints/array-map.d.ts

```diff
@@ -0,0 +1,15 @@
+/// <reference path="utils.d.ts" />
+
+interface ReadonlyArray<T> {
+  map<U>(
+    callbackfn: (value: T, index: number, array: readonly T[]) => U,
+    thisArg?: any,
+  ): { [K in keyof this]: U };
```

Comment:
```suggestion
  ): { readonly [K in keyof this]: U };
```

Reply:
hey @ahrjarrett what difference does this make? Could we test it?

Reply:
In my mind, if this doesn't seem to make a difference, that only happens to be true for the set of inputs we've tried it with.

Fixing the output of a readonly array's `map` operation to be readonly seems like a reasonable thing to do. It creates a cohesive API, its semantics are unambiguous, and it eliminates the possibility that one of the property modifiers might change out from under us.

Probably a moot point anyway, since I don't know if this addition is in keeping with the design goals of ts-reset. That said, I'm personally all for preserving structure like this where possible, so I went ahead and made the suggestion :)